### ISSUE TITLE
New versions without accepted pull requests with gzip requests

### ISSUE THREAD

**Issue by gawsoftpl** [SeldonIO/MLServer#1038]
Hi, 
My pull requests was accepted #950 but in new versions I can't see gzip compress for incoming requests:

In master branch is: https://raw.githubusercontent.com/SeldonIO/MLServer/master/mlserver/rest/requests.py
```python
from typing import Any
from gzip import decompress as gzip_decompress
from fastapi import Request as _Request

try:
    import orjson
except ImportError:
    orjson = None  # type: ignore


class Request(_Request):
    """
    Custom request class which uses `orjson` if present.
    Otherwise, it falls back to the standard FastAPI request.
    """

    async def body(self) -> bytes:
        if not hasattr(self, "_body"):
            body = await super().body()
            if "gzip" in self.headers.getlist("Content-Encoding"):
                body = gzip_decompress(body)
            self._body = body
        return self._body

    async def json(self) -> Any:
        if orjson is None:
            return await super().json()

        if not hasattr(self, "_json"):
            body = await self.body()
            self._json = orjson.loads(body)

        return self._json
```
in Version 1.24 requests.py have value:
```python
from typing import Any

from fastapi import Request as _Request

try:
    import orjson
except ImportError:
    orjson = None  # type: ignore


class Request(_Request):
    """
    Custom request class which uses `orjson` if present.
    Otherwise, it falls back to the standard FastAPI request.
    """

    async def json(self) -> Any:
        if orjson is None:
            return await super().json()

        if not hasattr(self, "_json"):
            body = await self.body()
            self._json = orjson.loads(body)

        return self._json
```

**Comment by gawsoftpl** [comment]
I see that there is multiple branches 1.2.x and 2.x.x
I will create pull requests for those branches

**Comment by gawsoftpl** [comment]
Added pull requests for this #1041 #1040

**Comment by adriangonz** [comment]
Hey @gawsoftpl ,

These branches are internal branches used to prepare releases.  PRs should only be merged against `master`. 

Since #950 was merged after the `1.2.x` release was out, it will need to wait until the `1.3.x` release. In the meantime, if you want to try it out, you should be able to use our dev releases (e.g. `1.3.0.dev3`).
